example/accel/trackingmanager-offload.cc

```diff
@@ -0,0 +1,209 @@
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file example/accel/trackingmanager-offload.cc
+//---------------------------------------------------------------------------//
+
+#include <algorithm>
+#include <iterator>
+#include <type_traits>
+#include <FTFP_BERT.hh>
+#include <G4Box.hh>
+#include <G4Electron.hh>
+#include <G4EmStandardPhysics.hh>
+#include <G4Gamma.hh>
+#include <G4LogicalVolume.hh>
+#include <G4Material.hh>
+#include <G4PVPlacement.hh>
+#include <G4ParticleDefinition.hh>
+#include <G4ParticleGun.hh>
+#include <G4ParticleTable.hh>
+#include <G4Positron.hh>
+#include <G4Region.hh>
+#include <G4RegionStore.hh>
+#include <G4RunManagerFactory.hh>
+#include <G4SystemOfUnits.hh>
+#include <G4Threading.hh>
+#include <G4ThreeVector.hh>
+#include <G4Track.hh>
+#include <G4TrackStatus.hh>
+#include <G4Types.hh>
+#include <G4UserEventAction.hh>
+#include <G4UserRunAction.hh>
+#include <G4UserTrackingAction.hh>
+#include <G4VUserActionInitialization.hh>
+#include <G4VUserDetectorConstruction.hh>
+#include <G4VUserPrimaryGeneratorAction.hh>
+#include <accel/AlongStepFactory.hh>
+#include <accel/LocalTransporter.hh>
+#include <accel/SetupOptions.hh>
+#include <accel/SharedParams.hh>
+#include <accel/SimpleOffload.hh>
+#include <accel/TrackingManagerOffload.hh>
+#include <corecel/Macros.hh>
+#include <corecel/io/Logger.hh>
+
+namespace
+{
+//---------------------------------------------------------------------------//
+// Global shared setup options
+celeritas::SetupOptions setup_options;
+// Shared data and GPU setup
+celeritas::SharedParams shared_params;
+// Thread-local transporter
+G4ThreadLocal celeritas::LocalTransporter local_transporter;
+
+// Simple interface to running celeritas
+G4ThreadLocal celeritas::SimpleOffload simple_offload;
+
+//---------------------------------------------------------------------------//
+class DetectorConstruction final : public G4VUserDetectorConstruction
+{
+  public:
+    DetectorConstruction()
+        : aluminum_{new G4Material{
+            "Aluminium", 13., 26.98 * g / mole, 2.700 * g / cm3}}
+    {
+        setup_options.make_along_step = celeritas::UniformAlongStepFactory();
+    }
+
+    G4VPhysicalVolume* Construct() final
+    {
+        CELER_LOG_LOCAL(status) << "Setting up detector";
+        auto* box = new G4Box("world", 1000 * cm, 1000 * cm, 1000 * cm);
+        auto* lv = new G4LogicalVolume(box, aluminum_, "world");
+        auto* pv = new G4PVPlacement(
+            0, G4ThreeVector{}, lv, "world", nullptr, false, 0);
+        return pv;
+    }
+
+  private:
+    G4Material* aluminum_;
+};
+
+//---------------------------------------------------------------------------//
+class EMPhysicsConstructor final : public G4EmStandardPhysics
+{
+  public:
+    using G4EmStandardPhysics::G4EmStandardPhysics;
+
+    void ConstructProcess() override
+    {
```

Comment:
The physics constructor class instance is "shared", but is `ConstructProcess` done thread-locally? For the sake of clarity for the example I'd add:

```suggestion
    {
        CELER_LOG_LOCAL(status) << "Setting up tracking manager offload";
```